sdk/src/wallet/wallet/builder.rs

```diff
@@ -166,27 +166,25 @@ impl WalletBuilder {
             self.secret_manager.replace(secret_manager);
         }
 
-        if self.coin_type.is_none() {
-            let coin_type =
-                read_manager_builder
-                    .and_then(|data| data.coin_type)
-                    .ok_or(crate::wallet::Error::MissingParameter(
-                        "coin_type (IOTA: 4218, Shimmer: 4219)",
-                    ))?;
+        let coin_type =
+            self.coin_type.map(Ok::<u32, crate::wallet::Error>).unwrap_or_else(|| {
```

Comment:
Is there a way to do the same without `.map(Ok::<u32, crate::wallet::Error>)` or matching and just passing the Some() value?

Reply:
`.ok_or_else` I believe is the right fn.

Reply:
That doesn't work because the error differs ```^ the trait `std::convert::From<std::result::Result<u32, _>>` is not implemented for `wallet::error::Error```

Could do this, but doesn't really look better
```Rust
        let coin_type = self
            .coin_type
            .or_else(|| {
                let coin_type = read_manager_builder.and_then(|data| data.coin_type);
                if let Some(coin_type) = coin_type {
                    // Update self so it gets stored
                    self.coin_type.replace(coin_type);
                }
                coin_type
            })
            .ok_or(crate::wallet::Error::MissingParameter(
                "coin_type (IOTA: 4218, Shimmer: 4219)",
            ))?;
```

Reply:
I prefer that tbh. More rusty